user.php

```diff
@@ -0,0 +1,108 @@
+<?
+
+
+class User{
+
+	public $db;
+
+	private $states = [
+		'NSW',
+		'ACT',
+		'VIC',
+		'QLD',
+		'TAS',
+		'WA'
+	]
+
+	function __construct()
+	{
+		$this->db = new mysqli('127.0.0.1', 'root', 'password', 'users');
+
+	}
+
+	function check_Postcode($postcode) {
+		//.... call some external api
+		$res = external_api_call();
+		if (!$res) {
+			throw new \Exception("invalid post code");
+		}
+		return true;
+	}
+
+
+	function register($name, $password, $email, $phone = null, $postal_address = null, $postalState = null
+			, $billing_address = null, $billingState = null)
+	{
+		global $validator;
+
+		//check for an existing account
+		$q = "SELECT * FROM user WHERE email = '$email'";
+		$result = $this->db->query($q);
+		while ($row = $result->fetch_assoc()) {
+		    if ($row['email'] == $email) {
+		    	return false;
+		    }
+		}
+
+		//Check for valid postal and billing states
+		$found = false;
+		foreach ($this->states as $s) {
+			if ($s == $postalState) {
+				$found = true;
+			}
+		}
+		if (!$found) {
+			throw \Exception('Invalid state');
+		}
+
+		$found = false;
+		foreach ($this->states as $s) {
+			if ($s == $billingState) {
+				$found = true;
+			}
+		}
+		if (!$found) {
+			throw \Exception('Invalid state');
+		}
+
+		//run additional validation checks
+		check_Postcode($postal_address);
```

Comment:
This is being called in the global namespace and it will cause a fatal error as it doesn't exist.

Reply:
it should be ```$this->check_Postcode($postal_address);```